File: 05. Sorting/SelectionSort.java
import java.util.Scanner;
/*
 //in selection sort here we first assume that the first element is the smallest one and then 
 we starting compare from index 1 i.e 2nd element .
 in bubble sort we swap while comapring if larger one is found before but in selection
 sort here we swap only once when one iteration complete .we just update the index or element
 of smaller element when found with the previous smaller one.
 and at the end of iteration we swap the current smaller element with the last smaller element.
 */
public class SelectionSort {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of array: ");
        int n=sc.nextInt();
        
        int arr[]=new int[n];
        System.out.println("Enter the size of array: ");
        for(int i=0; i<n; i++){
            arr[i]=sc.nextInt();
        }

        //selection sort
        for(int i=0; i<n-1; i++){
            int smallest = i;
            for(int j=i+1; j<n; j++){
                if(arr[smallest] > arr[j]){
                    smallest = j;
                }
            }
            int tmp=arr[smallest];
            arr[smallest]=arr[i];
            arr[i]=tmp;
        }
        printArray(arr);
    }
    public static void printArray(int arr[]){
        for(int i=0; i<arr.length; i++){
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
}


/*
 Enter the size of array:
5
Enter the size of array: 
3 0 -3 -5 33
-5 -3 0 3 33 

 */

File: 18. Javascript Progam/selectionSort.js
function selectionSort(data) {
    for (let i = 0; i < data.length; i++) {
        let min = i;
        for (let j = i + 1; j < data.length; j++) {
            if (data[j] < data[min]) {
                min = j
            }
            
        }
        if ( min !== i) [data[i], data[min]] = [data[min], data[i]]
    }
    return data
}
let arr = [28, 21, 4, 83, 5, 46, 95, 81, 23, 8, 53, 24, 9, 82, 98];
console.log(selectionSort(arr));
